### PR TITLE
Fix cannot find machine type after Terraform destroy

### PR DESCRIPTION
**Pull request by dancysoft**
Machine lists are available at the guest.arch level as well as within
each guest.arch.domain so getCanonicalMachineName must look in all of
these places.

Fixes https://github.com/dmacvicar/terraform-provider-libvirt/issues/691

### CODE REVIEW

**Comment by MalloZup** [comment]
@dancysoft hi. can you check why CI is failing for this PR ?

**Comment by dancysoft** [comment]
> @dancysoft hi. can you check why CI is failing for this PR ?

I took a look. This failure text follows. Tests pass on my devel machine so I'm inclined to blame this on something else unrelated to my changes.  I'll make a trivial change to the commit and repush to see how a retest goes.

=== RUN   TestAccLibvirtVolume_RepeatedName
--- FAIL: TestAccLibvirtVolume_RepeatedName (5.13s)
    testing.go:696: Error destroying resource! WARNING: Dangling resources
        may exist. The full state and error is shown below.
        
        Error: errors during apply: Can't retrieve volume /tmp/terraform-provider-libvirt-pool-ftntkyxpda/xwaydvqooo: virError(Code=50, Domain=18, Message='Storage volume not found: no storage vol with matching key /tmp/terraform-provider-libvirt-pool-ftntkyxpda/xwaydvqooo')
        
        State: libvirt_pool.ftntkyxpda:
          ID = 94e45a64-a02c-47a4-9f61-1a8ced8036ed
          provider = provider.libvirt
          allocation = 13512429568
          capacity = 72795082752
          name = ftntkyxpda
          path = /tmp/terraform-provider-libvirt-pool-ftntkyxpda
          type = dir
          xml.# = 0
        libvirt_volume.kuumqaqcpc:
          ID = /tmp/terraform-provider-libvirt-pool-ftntkyxpda/xwaydvqooo
          provider = provider.libvirt
          format = qcow2
          name = xwaydvqooo
          pool = ftntkyxpda
          size = 1073741824
          xml.# = 0

**Comment by dancysoft** [comment]
> @dancysoft hi. can you check why CI is failing for this PR ?

I made a trivial change to the commit message and repushed.  Travis CI is happy now. :-)

File: libvirt/utils_domain_def.go [hunk 1]
```diff
@@ -21,20 +21,39 @@ func getGuestForArchType(caps libvirtxml.Caps, arch string, virttype string) (li
 	return libvirtxml.CapsGuest{}, fmt.Errorf("[DEBUG] Could not find any guests for architecure type %s/%s", virttype, arch)
 }
 
+func huntForMachine(machines []libvirtxml.CapsGuestMachine, targetmachine string) string {
```
Comment (MalloZup) [inline libvirt/utils_domain_def.go:24]: `huntForMachine` is rather unusual name. 

I would call it `lookupForMachine`.

Comment (dancysoft) [reply libvirt/utils_domain_def.go]: I settled on `lookupMachine`.  Let me know if that is satisfactory.

**Comment by dancysoft** [comment]
Thanks for merging. It does look like the change is still having an impact on tests.  I'll see if I can figure out what's going on this week.

**Comment by dancysoft** [comment]
@moio Can we collaborate on dealing with the recent CI failures?

**Comment by moio** [comment]
I'd love to help, but to be completely honest I am in a high-stress period due to the Coronavirus emergency, so it'll take a bit of time before I can dedicate serious cycles to this project.

@MalloZup maybe you can help here?

**Comment by MalloZup** [comment]
@moio   yes sure I can help here no worry. take care :+1: 

@dancysoft we can create a separate issue regarding this I will ping you there.

**Pull request merged**